hs_geographic_feature_resource/receivers.py

```diff
@@ -312,50 +316,55 @@ def validate_form(request, element_name):
 def geofeature_pre_delete_file_from_resource(sender, **kwargs):
 
     res_obj = kwargs['resource']
-    del_file = kwargs['file']
-    one_file_removed = True
-    all_file_removed = False
-    del_res_fname = get_resource_file_name(del_file)
+    res_id = res_obj.short_id
+    try:
+        del_file = kwargs['file']
+        one_file_removed = True
+        all_file_removed = False
+        del_res_fname = get_resource_file_name(del_file)
+
+        ori_file_info = res_obj.metadata.originalfileinfo.all().first()
+        if ori_file_info.fileType == "SHP" or ori_file_info.fileType == "ZSHP":
+            del_f_fullname = del_res_fname[del_res_fname.rfind('/')+1:].lower()
+            del_f_name, del_f_ext = os.path.splitext(del_f_fullname)
+            if del_f_ext in [".shp", ".shx", ".dbf"]:
+                # The shp, shx or dbf files cannot be removed.
+                all_file_removed = True
```

Comment:
These two variables appear to be mutually exclusive, so you could get by with just `all_file_removed`.